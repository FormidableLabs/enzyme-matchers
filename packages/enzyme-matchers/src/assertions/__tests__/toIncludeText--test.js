const { mount } = require('enzyme');
const React = require('react');

const toIncludeText = require('../toIncludeText');

function Fixture() {
  return (
    <div>
      <p id="full">Some important text</p>
    </div>
  );
}

describe('toIncludeText', () => {
  const wrapper = mount(<Fixture />).find('#full');
  const truthyResults = toIncludeText(wrapper, 'important');
  const falsyResults = toIncludeText(wrapper, 'nope');
  const undefinedResults = toIncludeText(wrapper, undefined);

  it('returns the pass flag properly', () => {
    expect(truthyResults.pass).toBeTruthy();
    expect(falsyResults.pass).toBeFalsy();
    expect(undefinedResults.pass).toBeFalsy();
  });

  it('returns the message with the proper pass/fail verbage', () => {
    expect(truthyResults.message).not.toContain('not');
    expect(falsyResults.message).toContain('not');
    expect(undefinedResults.message).toContain('not');
  });
});
