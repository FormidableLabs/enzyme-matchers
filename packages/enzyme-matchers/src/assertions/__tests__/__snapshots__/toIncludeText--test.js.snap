// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toIncludeText mount provides contextual information for the message (mount) 1`] = `
Object {
  "actual": "Actual HTML: \\"Some important text\\"",
  "expected": "Expected HTML: \\"important\\"",
}
`;

exports[`toIncludeText mount returns the message with the proper fail verbage (mount) 1`] = `"Expected <p> not to contain \\"important\\" but it did."`;

exports[`toIncludeText mount returns the message with the proper pass verbage (mount) 1`] = `"Expected <p> to contain \\"important\\" but it did not."`;

exports[`toIncludeText shallow provides contextual information for the message (shallow) 1`] = `
Object {
  "actual": "Actual HTML: \\"Some important text\\"",
  "expected": "Expected HTML: \\"important\\"",
}
`;

exports[`toIncludeText shallow returns the message with the proper fail verbage (shallow) 1`] = `"Expected <p> not to contain \\"important\\" but it did."`;

exports[`toIncludeText shallow returns the message with the proper pass verbage (shallow) 1`] = `"Expected <p> to contain \\"important\\" but it did not."`;
